Runtime/CesiumCameraController.cs

```diff
@@ -0,0 +1,1145 @@
+            }
+
+            float height, viewDistance;
+
+            // Raycast from the camera to the Earth's center and compute the distance.
+            // Ignore the result if the height is approximately 0.
+            if (!this.RaycastTowardsEarthCenter(out height) || height < 0.000001f)
+            {
+                overrideSpeed = false;
+                newSpeed = 0.0f;
+
+                return false;
+            }
+
+            // Also ignore the result if the speed will increase by too much at once.
+            // This can be an issue when 3D tiles are loaded/unloaded from the scene.
+            if (
+                this._maxSpeedPreMultiplier > 0.5f &&
+                (height / this._maxSpeedPreMultiplier) > 1000.0f)
+            {
+                overrideSpeed = false;
+                newSpeed = 0.0f;
+
+                return false;
+            }
+
+            // Raycast along the camera's view (forward) vector.
+            float raycastDistance = 
+                Mathf.Clamp(this._maxSpeed * 3.0f, 0.0f, this._maxRaycastDistance);
+
+            // If the raycast does not hit, then only override speed if the height
+            // is lower than the maximum threshold. Otherwise, if both raycasts hit,
+            // always override the speed.
+            if (!this.RaycastAlongForwardVector(raycastDistance, out viewDistance) ||
+                viewDistance < 0.000001f)
+            {
+                overrideSpeed = height <= this._dynamicSpeedMinHeight;
+            }
+            else
+            {
+                overrideSpeed = true;
+            }
+
+            // Set the speed to be the height of the camera from the Earth's center.
+            newSpeed = height;
+
+            return true;
+        }
+
+        private void ResetSpeedMultiplier()
+        {
+            this._speedMultiplier = 1.0f;
+        }
+
+        private void SetMaxSpeed(float speed)
+        {
+            float actualSpeed = this._maxSpeedCurve.Evaluate(speed);
+            this._maxSpeed = this._speedMultiplier * actualSpeed;
+            this._acceleration =
+                Mathf.Clamp(this._maxSpeed * 5.0f, 20000.0f, 10000000.0f);
+        }
+
+        private void UpdateDynamicSpeed()
+        {
+            bool overrideSpeed;
+            float newSpeed;
+            if (this.GetDynamicSpeed(out overrideSpeed, out newSpeed))
+            {
+                if (overrideSpeed || newSpeed >= this._maxSpeedPreMultiplier)
+                {
+                    this._maxSpeedPreMultiplier = newSpeed;
+                }
+            }
+
+            this.SetMaxSpeed(this._maxSpeedPreMultiplier);
+        }
+
+        #endregion
+
+        #region Fly-To helper functions
+
+        /// <summary>
+        /// Advance the camera flight based on the given time delta.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This function requires the CesiumGlobeAnchor to be valid. If it is not valid,
```

Comment:
Do you mean the `CesiumGeoreference`? I don't think the CesiumGlobeAnchor can be missing, because of the `RequireComponent` attribute.